content/assignments/minishell/checker/_test/run_test.sh

```diff
@@ -0,0 +1,252 @@
+
+# Specific test example
+execute_cmd()
+{
+	EXEC=$1
+	INPUT=$2
+	OUTPUT=$3
+	[ -z "$OUTPUT" ] && OUTPUT=$LOG_FILE
+	mkdir -p "${OUT_DIR}" && cd "${OUT_DIR}" &> "$LOG_FILE" || exit 1
+	timeout "$TEST_TIMEOUT" $BUFFERING_WRAPPER $EXEC < "${INPUT}" &> "$OUTPUT"
+	# Ocasionally, in a virtualized environment, the diff that compares the
+	# target implementation with the reference output starts before the
+	# target implementation has finished writing files to disk. Handle this
+	# by mandating that all I/O buffers be flushed.
+	sync
+	cd - &> "$LOG_FILE" || exit 1
+}
+
+
+generate_input()
+{
+	# generate the PROMPT for BASH
+	sed 's/\(.*\)/printf \"> \"; \1/' "${IN_FILE}" > "${REF_FILE}"
+}
+
+# checks the output of the commands
+test_output()
+{
+	init_test
+
+	generate_input
+
+	# commands to execute the test
+	execute_cmd "$ref_name" "../${REF_FILE}" "${OUT_FILE}"
+	# move OUT_DIR in order to preserve the pwd output
+	mv "${OUT_DIR}" "${REF_DIR}"
+	execute_cmd "$exec_name" "../${IN_FILE}" "${OUT_FILE}"
+
+	# test output
+	basic_test diff -r -ui -x "$VALGRIND_LOG" "${REF_DIR}" "${OUT_DIR}"
+
+	cleanup_test
+}
+
+# tests common commands
+_test_common()
+{
+	init_test
+
+	# commands to execute the test
+	execute_cmd "$1" "../${IN_FILE}"
+	# move OUT_DIR in order to preserve the pwd output
+	mv "${OUT_DIR}" "${REF_DIR}"
+	execute_cmd "$exec_name" "../${IN_FILE}"
+
+	# test output
+	basic_test diff -r -ui -x "$VALGRIND_LOG" "${REF_DIR}" "${OUT_DIR}"
+
+	cleanup_test
+}
+
+# default common test
+test_common()
+{
+	_test_common "$ref_name"
+}
+
+# tests common commands using alternate ref
+test_common_alt()
+{
+	if [ "$OS_PLATFORM" == "win" ]; then
+		_test_common "$ref_name_alt"
+	else
+		_test_common "$ref_name"
+	fi
+}
+
+# test 18
+test_exec_failed()
+{
+	init_test
+
+	# commands to execute the test
+	execute_cmd "$exec_name" "../${IN_FILE}" "${REF_FILE}"
+
+	# test output
+	basic_test diff -r -uib "${REF_FILE}" "${OUT_DIR}/${REF_FILE}"
+
+	cleanup_test
+}
+
+
+test_fun_array=(								\
+	test_coding_style	"Sources check"				5	\
+	test_output		"Testing commands without arguments"	3	\
+	test_output		"Testing commands with arguments"	2	\
+	test_common		"Testing simple redirect operators"	5	\
+	test_common		"Testing append redirect operators"	5	\
+	test_common		"Testing current directory"		5	\
+	test_common		"Testing conditional operators"		5	\
+	test_common		"Testing sequential commands"		3	\
+	test_common		"Testing environment variables"		5	\
+	test_common		"Testing single pipe"			5	\
+	test_common		"Testing multiple pipes"		10	\
+	test_common		"Testing variables and redirect"	5	\
+	test_common		"Testing overwritten variables"		2	\
+	test_common		"Testing all operators"			2	\
+	test_common_alt		"Testing parallel operator"		10	\
+	test_common_alt		"Testing big file"			5	\
+	test_common_alt		"Testing sleep command"			7	\
+	test_common_alt		"Testing fscanf function"		7	\
+	test_exec_failed	"Testing unknown command"		4	\
+)
```

Comment:
The total score is 95. For the previous assignments, the total was 90. I think we should stick to that.

Reply:
IIRC, the discussion was to award the last 10 points (from 90 to 100) after checking the assignment manually. This means going beyond checkpatch and checking for code coherence, modularity etc.

Reply:
Yes, you are right. But I don't understand your suggestion.
How do you want to handle this?
Exclude checkpatch from checker? 
Let it to run only with a special flag without any scoring?
Adjust the score to 90/100 and still exclude checkpatch? Adjust the score to 90/90?

Reply:
Keep checkpatch in the checker, but lower the score to 90/100, like the other assignments.